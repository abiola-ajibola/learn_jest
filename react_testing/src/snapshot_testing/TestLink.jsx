import { useState } from "react";

export default function TestLink({ url, children }) {
  const [hover, setHover] = useState("normal");

  const handleMouseEnter = () => setHover("hover");
  const handleMouseLeave = () => setHover("normal");

  return (
    <a
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
      href={url}
      className={hover}
    >
      {children}
    </a>
  );
}
