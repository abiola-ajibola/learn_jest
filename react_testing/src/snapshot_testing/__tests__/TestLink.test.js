import React from "react"
import renderer from "react-test-renderer"
import TestLink from "../TestLink"

test("Change className value on hover", () => {
    const componenent = renderer.create(<TestLink url={"https://google.com"}>Google</TestLink>)
    let tree = componenent.toJSON()
    expect(tree).toMatchSnapshot()

    tree.props.onMouseEnter()
    tree = componenent.toJSON()
    expect(tree).toMatchSnapshot()

    tree.props.onMouseLeave()
    tree = componenent.toJSON()
    expect(tree).toMatchSnapshot()
})