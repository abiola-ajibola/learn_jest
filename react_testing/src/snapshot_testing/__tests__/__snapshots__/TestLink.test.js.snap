// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change className value on hover 1`] = `
<a
  className="normal"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Change className value on hover 2`] = `
<a
  className="hover"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;

exports[`Change className value on hover 3`] = `
<a
  className="normal"
  href="https://google.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Google
</a>
`;
